flocker/node/agents/ebs.py

```diff
@@ -0,0 +1,213 @@
+        size=int(GB(ebs_volume.size).to_Byte().value),
+        host=None,
+        dataset_id=UUID(ebs_volume.tags[DATASET_ID_LABEL])
+    )
+
+
+def _wait_for_volume(expected_volume,
+                     expected_status=u'available',
+                     time_limit=60):
+    """
+    Helper function to wait for up to 60s for given volume
+    to be in 'available' state.
+    """
+    start_time = time.time()
+    expected_volume.update()
+    while expected_volume.status != expected_status:
+        elapsed_time = time.time() - start_time
+        if elapsed_time < time_limit:
+            time.sleep(0.1)
+            expected_volume.update()
+        else:
+            raise Exception(
+                'Timed out while waiting for volume. '
+                'Expected Volume: {!r}, '
+                'Expected Status: {!r}, '
+                'Actual Status: {!r}, '
+                'Elapsed Time: {!r}, '
+                'Time Limit: {!r}.'.format(
+                    expected_volume, expected_status,
+                    expected_volume.status, elapsed_time,
+                    time_limit
+                )
+            )
+
+
+def _is_cluster_volume(cluster_id, ebs_volume):
+    """
+    Helper function to check if given volume belongs to
+    given cluster.
+    """
+    actual_cluster_id = ebs_volume.tags.get(CLUSTER_ID_LABEL)
+    if actual_cluster_id is not None:
+        actual_cluster_id = UUID(actual_cluster_id)
+        if actual_cluster_id == cluster_id:
+            return True
+    return False
+
+
+@implementer(IBlockDeviceAPI)
+class EBSBlockDeviceAPI(object):
+    """
+    An EBS implementation of ``IBlockDeviceAPI`` which creates
+    block devices in an EC2 cluster using Boto APIs.
+    """
+    def __init__(self, ec2_client, cluster_id):
+        """
+        Initialize EBS block device API instance.
+
+        :param ``_EC2`` ec2_client: A record of EC2 connection and zone.
+        :param String cluster_id: UUID of cluster for this
+            API instance.
+        """
+        self.connection = ec2_client.connection
+        self.zone = ec2_client.zone
+        self.cluster_id = cluster_id
+
+    def _blockdevicevolume_from_ebs_volume(self, ebs_volume):
+        """
+        Helper function to convert Volume information from
+        EBS format to Flocker block device format.
+        """
+        return BlockDeviceVolume(
+            blockdevice_id=unicode(ebs_volume.id),
+            size=int(GB(ebs_volume.size).to_Byte().value),
+            host=None,
+            dataset_id=UUID(ebs_volume.tags[DATASET_ID_LABEL])
+        )
+
+    def create_volume(self, dataset_id, size):
+        """
+        Create a volume on EBS. Store Flocker-specific
+        {metadata version, cluster id, dataset id} for the volume
+        as volume tag data.
+        Open issues: https://clusterhq.atlassian.net/browse/FLOC-1792
+
+        :param String dataset_id: Dataset_id for the volume.
```

Comment:
String > str

But actually another rule-of-thumb is that we don't bother documenting parameters and return types when they're already documented in a zope interface....which is the case here. 
So just leave the implementation specific notes.